state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -784,61 +709,64 @@ fn retry_with_threshold_signing_still_allows_late_success_witness_second_attempt
+	new_test_ext()
+		.execute_with(|| {
+			MockBroadcastBarriers::set(ChainChoice::Polkadot);
 
-		// create and sign 3 txs that are then ready for broadcast
+			let (tx_out_id1, api_call1) = api_call(1);
+			let (tx_out_id2, api_call2) = api_call(2);
+			let (tx_out_id3, api_call3) = api_call(3);
 
-		let broadcast_id_1 = initiate_and_sign_broadcast(&api_call1, TxType::Normal);
-		// tx1 emits broadcast request
-		assert_transaction_broadcast_request_event(broadcast_id_1, 0, tx_out_id1);
+			// create and sign 3 txs that are then ready for broadcast
+			let broadcast_id_1 = initiate_and_sign_broadcast(&api_call1, TxType::Normal);
+			// tx1 emits broadcast request
+			assert_transaction_broadcast_request_event(broadcast_id_1, tx_out_id1);
 
-		let broadcast_id_2 = initiate_and_sign_broadcast(&api_call2, TxType::Rotation);
-		// tx2 emits broadcast request and also pauses any further new broadcast requests
-		assert_transaction_broadcast_request_event(broadcast_id_2, 0, tx_out_id2);
+			let broadcast_id_2 = initiate_and_sign_broadcast(&api_call2, TxType::Rotation);
+			// tx2 emits broadcast request and also pauses any further new broadcast requests
+			assert_transaction_broadcast_request_event(broadcast_id_2, tx_out_id2);
 
-		let broadcast_id_3 = initiate_and_sign_broadcast(&api_call3, TxType::Normal);
+			let broadcast_id_3 = initiate_and_sign_broadcast(&api_call3, TxType::Normal);
 
-		// tx3 is ready for broadcast but since there is a broadcast pause, broadcast request is
-		// not issued, the broadcast is rescheduled instead.
-		System::assert_last_event(RuntimeEvent::Broadcaster(
-			crate::Event::<Test, Instance1>::BroadcastRetryScheduled {
-				broadcast_attempt_id: BroadcastAttemptId {
+			// tx3 is ready for broadcast but since there is a broadcast pause, broadcast request is
+			// not issued, the broadcast is rescheduled instead.
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastRetryScheduled {
 					broadcast_id: broadcast_id_3,
-					attempt_count: 0,
+					retry_block: System::block_number() + 1,
 				},
-			},
-		));
-
-		// report successful broadcast of tx1
-		witness_broadcast(tx_out_id1);
+			));
 
-		// tx3 should still not be broadcasted because the blocking tx (tx2) has still not
-		// succeeded.
-		Broadcaster::on_idle(0, LARGE_EXCESS_WEIGHT);
-		assert_eq!(
-			BroadcastRetryQueue::<Test, Instance1>::get()[0]
-				.broadcast_attempt_id
-				.broadcast_id,
-			broadcast_id_3
-		);
+			// report successful broadcast of tx1
+			witness_broadcast(tx_out_id1);
 
-		// Now tx2 succeeds which should allow tx3 to be broadcast
-		witness_broadcast(tx_out_id2);
-		Broadcaster::on_idle(0, LARGE_EXCESS_WEIGHT);
+			// tx3 should still not be broadcasted because the blocking tx (tx2) has still not
+			// succeeded.
+			(tx_out_id2, tx_out_id3, broadcast_id_3)
+		})
+		.then_execute_at_next_block(|(tx_out_id2, tx_out_id3, broadcast_id_3)| {
```

Comment:
we should add a util that does this pass through without having to pass everything through manually - we do this in a number of places (would be in a separate PR ofc)

Reply:
https://linear.app/chainflip/issue/PRO-1090/context-passing-in-test-harness